lib/cc/analyzer/engines_config_builder.rb

```diff
@@ -29,12 +29,22 @@ def run
 
       private
 
-      attr_reader :include_paths
+      attr_reader :include_paths_builder
+
+      def include_paths(engine_excludes)
+        if @requested_paths.empty?
+          IncludePathsBuilder.new(global_excludes).build(engine_excludes)
+        else
+          IncludePathsBuilder.new([], @requested_paths).build
```

Comment:
I might be missing the complete flow of things, but aren't we going to honor `excludes` from `.codeclimate.yml` unless `--ignore-excludes` gets set? and if so, would we do that here?